### DISCUSSION TITLE
Create a re-usable, predefined hash filter

### DISCUSSION THREAD

**Discussion by ximus** [AaronLasseigne/active_interaction#532] [Q&A] [answered]
Hi there,

I've got a piece of data input that gets repeated around various commands. It's in the form of a Hash and always has the same structure. Is there recommended way to define the structure of my hash input once and reference it from any command?

**example**: a data type representing a "quantity". a `quantity` is defined by a `unit` (kg, litres, etc ...) and an `amount`, represented by a hash from my web controller params.

I'm wondering if I could do something as such, somehow:

```ruby
    class OrderFood < ActiveInteraction::Base
      string :name
      quantity :qty # <-- quantity is a predefined hash filter

      def execute
        puts "ordering #{qty[:amount]} #{qty[:unit]} of #{name}"
      end
    end
```

also I might want to nest this `quantity` data type within `hash` or `array` types, so I can't just define a class method on a base command class.

Thanks for the lovely library, you rock.

**Comment by AaronLasseigne** [comment, marked as answer]
Thanks!

What you want can be done but it's not the easiest thing. The problem is there's no documentation for doing it and I haven't committed to a public interface for creating custom filters. It does seem like something that would be useful though. I'll think about what it would take to document it. In the meantime you'll have to dig a bit around in the existing filters.
